android/src/main/java/io/invertase/firebase/links/RNFirebaseLinks.java

```diff
@@ -43,37 +42,61 @@ public String getName() {
   }
 
   @ReactMethod
-  public void getInitialLink(Promise promise) {
-    promise.resolve(initialLink);
+  public void getInitialLink(final Promise promise) {
+    Activity activity = getCurrentActivity();
+    if (activity != null) {
+      FirebaseDynamicLinks.getInstance()
+        .getDynamicLink(activity.getIntent())
+        .addOnSuccessListener(activity, new OnSuccessListener<PendingDynamicLinkData>() {
+          @Override
+          public void onSuccess(PendingDynamicLinkData pendingDynamicLinkData) {
+            // Get deep link from result (may be null if no link is found or run on the same intent again)
+            if (pendingDynamicLinkData != null) {
```

Comment:
This bit of code. Shouldn't we return the `initialLink` if we have one anyway (if we have an `initialLink` why do we need to resolve the intent again)?